parsec/mca/termdet/local/termdet_local_module.c

```diff
@@ -128,11 +160,11 @@ static int parsec_termdet_local_taskpool_ready(parsec_taskpool_t *tp)
     assert( tp->tdm.monitor == PARSEC_TERMDET_LOCAL_NOT_READY );
     parsec_atomic_cas_ptr(&tp->tdm.monitor, PARSEC_TERMDET_LOCAL_NOT_READY, PARSEC_TERMDET_LOCAL_BUSY);
     PARSEC_DEBUG_VERBOSE(10, parsec_debug_output, "TERMDET-LOCAL:\tTASKPOOL %p READY", tp);
-
+    PARSEC_OBJ_RETAIN(tp);
```

Comment:
where is the corresponding release for this?

Reply:
The `PARSEC_OBJ_RELEASE(tp)` corresponding to `PARSEC_OBJ_RETAIN(tp)` in `parsec_termdet_local_taskpool_ready` is at line 153 (https://github.com/ICLDisco/parsec/pull/698/files#diff-c4a4ca9c4b7ae4be5d5f60191e9f84074e2bd5533851a68ebca23afdc8545f02R153): once a taskpool is marked as `READY`, it must be terminated. That's why the check in `parsec_termdet_local_unmonitor` checks that the taskpool is either `UNMONITORED` (we haven't started it with `parsec_termdet_local_taskpool_ready`) or `TERMINATED` (we have started it, and it reached completion).

Reply:
thanks I missed it